src/lib.rs

```diff
@@ -1,35 +1,109 @@
-//! This is the documentation for the rust-abci crate.
-#![feature(ptr_internals)]
+//! # Tendermint ABCI library for Rust
+//!
+//! This library provides an application Trait and TCP server for implementing Tendemint ABCI
+//! application in Rust.  The Application Trait provides default implementations for each callback
+//! to simplify development.
+//!
+//! ## Example
+//!
+//! Here's a simple example that communicates with Tendermint. Defaults callbacks are handled by
+//! the Trait.  The app doesn't do any actual processing on a transaction.
+//!
+//! ```
+//! struct EmptyApp;
+//!
+//! impl abci::Application for EmptyApp {}
+//!
+//! abci::run_local(EmptyApp);
+//!```
+//!
+use std::net::SocketAddr;
+
 extern crate bytes;
-extern crate protobuf;
 extern crate integer_encoding;
+extern crate protobuf;
 
-mod tcpserver;
-pub mod server;
+mod server;
 pub mod types;
 
+use server::serve;
 pub use types::*;
 
-pub trait Application : Sync {
+// Main Trait for ABCI application. Provides generic responses for all callbacks
```

Comment:
Make these and all other description comments triple slashes (`///`), so we can see them in the documentation.